### ISSUE TITLE
VFS Capability detection

### ISSUE THREAD

**Issue by andersevenrud** [os-js/OS.js#804]
VFS could benefit from capability detection so that clients is aware of what a filesystem source is capable of.

This would allow for adding extensions on the client that are context aware, like pagination and archives.

Split out from:

* https://github.com/os-js/osjs-client/pull/165
* https://github.com/os-js/osjs-server/pull/56

Relevant:

* https://github.com/os-js/osjs-client/pull/182
* https://github.com/os-js/osjs-server/pull/63
* https://github.com/os-js/osjs-client/pull/164

**Comment by andersevenrud** [comment]
As discussed, @mahsashadi will do some work related to this in the following stages:

1. URL serialization
2. Client option filtering
3. Capability detection
4. Other improvements

**Comment by andersevenrud** [comment]
Also, @ajmeese7 [wanted to have a look](https://github.com/os-js/osjs-filemanager-application/issues/39#issuecomment-1198676600) at archive extensions.

Thanks a bunch!

**Comment by mahsashadi** [comment]
I continue with **capability detection** of VFS:
https://github.com/os-js/osjs-client/pull/186
https://github.com/os-js/osjs-server/pull/65

These changes are considered that added method has some argument. If you still have the idea of [capability caching](https://github.com/os-js/osjs-filemanager-application/pull/37#issuecomment-980382961), maybe we should create another PR that changes the api to support methods without arguments. 
I remember I worked on [this issue](https://github.com/os-js/OS.js/issues/788) but did not come up with solution.

**Comment by andersevenrud** [comment]
@mahsashadi Yeah, let's tackle the caching in a separate issue. It needs further discussion. I have some ideas about this already, but let's take it when the time is right.

**Comment by mahsashadi** [comment]
I continue with **adding total values to VFS stat api** (needed for pagination in file manager):
https://github.com/os-js/osjs-server/pull/66

**Comment by mahsashadi** [comment]
I made another PR for VFS context-aware sorting in client-side.
https://github.com/os-js/osjs-client/pull/189
